cardano-cli/src/Cardano/CLI/Common/Parsers.hs

```diff
@@ -92,15 +92,19 @@ pEpochSlots =
     , Opt.showDefault
     ]
 
-pSocketPath :: Parser (Maybe SocketPath)
-pSocketPath =
-  optional $ fmap SocketPath $
-    Opt.strOption $ mconcat
-      [ Opt.long "socket-path"
-      , Opt.metavar "SOCKET_PATH"
-      , Opt.help $ mconcat
-        [ "Path to the node socket.  This overrides the CARDANO_NODE_SOCKET_PATH "
-        , "environment variable"
+pSocketPath :: EnvCli -> Parser SocketPath
```

Comment:
Why not leverage `Alternative`?:
```
pSocketPath :: EnvCli -> Parser SocketPath
pSocketPath envCli =
  let opts = mconcat
              [ Opt.long "socket-path"
              , Opt.metavar "SOCKET_PATH"
              , Opt.help $ mconcat
                [ "Path to the node socket.  This overrides the CARDANO_NODE_SOCKET_PATH "
                , "environment variable"
                ]
              , Opt.completer (Opt.bashCompleter "file")
              ]
  in fmap SocketPath (Opt.strOption opts) <|>
     -- Default to the socket path specified by the environment variable if it is available.
     pure (maybe (error "No socket path specified via cli nor env var") SocketPath $ envCliSocketPath envCli)
  ```
  
  And then we can `error` immediately rather than having some obscure networking error.

Reply:
We already use `Alternative` via `asum`.  If `CARDANO_NODE_SOCKET_PATH` is not supplied, the code as it currently is will error immediately because `--socket-path` is the only `Alternative` in this case.  The CLI parsing will fail because `--socket-path` is not supplied and it is the only alternative.  You only get the fallback alternative if `CARDANO_NODE_SOCKET_PATH` is supplied, in which case the socket path provided by the environment variable is the default and `--socket-path` will override.

Reply:
I updated the CLI help message to be clearer.